### PR TITLE
Update/add GA events

### PR DESCRIPTION
**Pull request by erikphansen**
## Description
Adding/updating the GA events that are fired from each form's intro page when users are both logged in and logged out.

## Testing done
Confirmed the correct GA events are fired by temporarily inserting a `console.log()` in the `recordEvent()` method.

## Acceptance criteria
- [ ]

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/platform/forms/save-in-progress/FormStartControls.jsx [hunk 1]
```diff
@@ -26,12 +27,10 @@ class FormStartControls extends React.Component {
 
   captureAnalytics = () =>
     this.props.gaStartEventName &&
-    window.dataLayer.push({ event: this.props.gaStartEventName });
```
Comment (erikphansen) [inline src/platform/forms/save-in-progress/FormStartControls.jsx]: `recordEvent()` is just a wrapper around `window.dataLayer.push()` with a check to make sure that it exists before calling it.

File: src/platform/forms/save-in-progress/FormStartControls.jsx [hunk 2]
```diff
@@ -26,12 +27,10 @@ class FormStartControls extends React.Component {
...
+    recordEvent({ event: this.props.gaStartEventName });
 
   handleLoadPrefill = () => {
     this.captureAnalytics();
-    // temp hack to fix the fact that the START button doesn't work
-    // this.goToBeginning();
```
Comment (erikphansen) [inline src/platform/forms/save-in-progress/FormStartControls.jsx]: this should have never been committed; I'd added it a while ago while making other changes.

File: src/platform/forms/save-in-progress/FormStartControls.jsx [hunk 3]
```diff
@@ -131,14 +130,17 @@ FormStartControls.propTypes = {
   removeInProgressForm: PropTypes.func.isRequired,
   router: PropTypes.object.isRequired,
   formSaved: PropTypes.bool.isRequired,
-  // prefillAvailable = whether the form can be pre-filled
```
Comment (erikphansen) [inline src/platform/forms/save-in-progress/FormStartControls.jsx]: I don't think this comment is require

File: src/platform/forms/save-in-progress/FormStartControls.jsx [hunk 4]
```diff
@@ -131,14 +130,17 @@ FormStartControls.propTypes = {
...
   prefillAvailable: PropTypes.bool.isRequired,
   startPage: PropTypes.string.isRequired,
   startText: PropTypes.string,
   resumeOnly: PropTypes.bool,
   gaStartEventName: PropTypes.string,
 };
 
+FormStartControls.defaultProps = {
+  gaStartEventName: 'login-successful-start-form',
```
Comment (erikphansen) [inline src/platform/forms/save-in-progress/FormStartControls.jsx]: It looks as though the `gaStartEventName` prop was added by @cvalarida seven months ago but we never actually use it anywhere. Adding this default value here handles the `CTA 3` case as outlined here: https://github.com/department-of-veterans-affairs/vets.gov-team/issues/16807.

We can easily override it anywhere we use a `SaveInProgressIntro` component by adding a `gaStartEventName` prop on the component and `SaveInProgressIntro` just passes the value down to the `FormStartControls`

File: src/platform/forms/save-in-progress/SaveInProgressIntro.jsx [hunk 5]
```diff
@@ -207,11 +208,12 @@ class SaveInProgressIntro extends React.Component {
   };
 
   goToBeginning = () => {
+    recordEvent({ event: 'no-login-start-form' });
```
Comment (erikphansen) [inline src/platform/forms/save-in-progress/SaveInProgressIntro.jsx]: This handles the `CTA 2` case defined here: https://github.com/department-of-veterans-affairs/vets.gov-team/issues/16807.

File: src/platform/forms/save-in-progress/SaveInProgressIntro.jsx [hunk 6]
```diff
@@ -207,11 +208,12 @@ class SaveInProgressIntro extends React.Component {
...
     this.props.router.push(this.getStartPage());
   };
 
   openLoginModal = () => {
-    this.props.toggleLoginModal(true);
+    this.props.toggleLoginModal(true, 'cta-form');
```
Comment (erikphansen) [inline src/platform/forms/save-in-progress/SaveInProgressIntro.jsx]: This handles the `CTA 1` case defined here: https://github.com/department-of-veterans-affairs/vets.gov-team/issues/16807.

**Review by cvalarida — APPROVED** [review]
Seems like an odd event name for that particular action, but it looks like it matches what's in the connected ticket, so :man_shrugging:

Actually using GA to find useful data baffles me, so I don't pretend to know anything on that front. :sweat_smile:  

Looks good! :shipit:

**Pull request merged**